arviz/data/inference_data.py

```diff
@@ -173,6 +175,53 @@ def __delattr__(self, group):
     def _groups_all(self):
         return self._groups + self._groups_warmup
 
+    def extend_xr_method(func):  # pylint: disable=no-self-argument
```

Comment:
why this method doesn't need a self argument?

Reply:
Is this function meant for general users or only developers / advanced users?

Reply:
I would make it private

Reply:
Done. Thanks!!